### ISSUE TITLE
idea: add tasker plugin integration

### ISSUE THREAD

**Issue by noahstreller** [Myzel394/Alibi#15]
Using Tasker or Macrodroid to start, stop or save a recording session could be incredibly useful

**Comment by Myzel394** [comment]
Yes, I mentioned something similar already in #8, I definitely want to add that! I will probably not be able to do it in the near future, but I will keep you updated here! :)

**Comment by Myzel394** [comment]
Okay so I just got an idea that I think might be even more useful than tasker / Automate.

:star: Plugins :star: - basically you'd be able to write your own Alibi plugins which would then be able to start / stop / pause etc the recording. I'd provide some builtin plugins so that basic features are met :)
I think plugins would be a bit better option since you have more control this way, Alibi won't rely on another app for automation and so you won't have to install a non-foss automation app :D

We could add plugins such as:
* Automatically save recording if a sudden loud noise happens
* Automatically stop recording after 5 hours
* Automatically stop recording if there hasn't been any noise for 30 minutes straight

these are just some examples, with plugins you'd be able to completely control the behavior!

However, I'm not sure if we can add automatically starting recordings due to Android's security. I'll have to look that up.

What do you think about that? This is quite a big feature, so I'd like to get your opinion on that first :D @notkudu and @sjbr your related feature requests #84 and #73 could also be implemented with a plugin this way - what do you think?

**Comment by notkudu** [comment]
> Okay so I just got an idea that I think might be even more useful than tasker / Automate.
> 
> ⭐ Plugins ⭐ - basically you'd be able to write your own Alibi plugins which would then be able to start / stop / pause etc the recording. I'd provide some builtin plugins so that basic features are met :) I think plugins would be a bit better option since you have more control this way, Alibi won't rely on another app for automation and so you won't have to install a non-foss automation app :D
> 
> We could add plugins such as:
> 
> * Automatically save recording if a sudden loud noise happens
> * Automatically stop recording after 5 hours
> * Automatically stop recording if there hasn't been any noise for 30 minutes straight
> 
> these are just some examples, with plugins you'd be able to completely control the behavior!
> 
> However, I'm not sure if we can add automatically starting recordings due to Android's security. I'll have to look that up.
> 
> What do you think about that? This is quite a big feature, so I'd like to get your opinion on that first :D @notkudu and @sjbr your related feature requests #84 and #73 could also be implemented with a plugin this way - what do you think?

that does sound like a good solution

**Comment by xandro0777** [comment]
There are 3 FOSS automation Apps in F-Droid - Easer, Automation and somewhat different Termux. Your idea with plugins is interesting but Easier/Automation can do many other things that would be a waste to reimplements in this App.

**Comment by WinnyKing57** [comment]
Hello,

Integration of the Smartlife API or the ability to add Smartlife/Tuya devices would be great.

**Comment by nyarau** [comment]
Have you considered supporting Android [Intent Broadcast](https://developer.android.com/develop/background-work/background-tasks/broadcasts)?
It's widely supported by automation apps:
- Tasker: https://tasker.joaoapps.com/userguide/en/intents.html
- Automate: https://llamalab.com/automate/doc/block/broadcast_send.html
- Easer: https://me.ryey.icu/Easer/en/FEATURES.html#event
- Termux: https://github.com/termux/TermuxAm

One caveat is that since any app can send a broadcast, you need to implement some sort of filter/whitelist if you don't trust local apps.
